ansible/site.yml

```diff
@@ -20,131 +13,173 @@
-          Application: "BlueButton"
-        instance_tags:
-          Name: bluebutton-stress-master
-          Application: "BlueButton"
-          CreatedBy: "{{ whoami.stdout }}"
-      register: ec2_master
-    
-    - name: EC2 - Provision 'bluebutton-stress' slaves
-      ec2:
-        key_name: "{{ ec2_key_name }}"
-        group:
-          - default
-          - ssh-all
-        instance_type: "{{ ec2_instance_type }}"
-        image: "{{ ami_id_ubuntu }}"
-        region: "{{ region }}"
-        wait: true
-        # The default of 300s here wasn't cutting it for 8 slaves.
-        wait_timeout: 600
-        exact_count: "{{ num_slaves }}"
-        count_tag:
-          Name: bluebutton-stress-slave
-          Application: "BlueButton"
-        instance_tags:
-          Name: bluebutton-stress-slave
-          Application: "BlueButton"
-          CreatedBy: "{{ whoami.stdout }}"
-      register: ec2_slaves
-    
-    - name: Create inventory group 'master'
-      add_host: hostname={{ item.public_ip }} groupname=master
-      with_items: ec2_master.instances
-    
-    - name: Create inventory group 'slaves'
-      add_host: hostname={{ item.public_ip }} groupname=slaves
-      with_items: ec2_slaves.instances
+    - name: Create inventory group 'servers'
+      add_host: hostname={{ item }} groupname=servers
+      with_items: jmeter-server 
     
-    - name: Wait for SSH to come up
-      wait_for: host={{ item.public_dns_name }} port=22 timeout=320 state=started
-      with_flattened:
-        - ec2_master.instances
-        - ec2_slaves.instances
-      
-- name: Configure instance(s)
-  hosts: master:slaves
+#    - name: Wait for SSH to come up
+#      wait_for: host={{ item.public_dns_name }} port=22 timeout=320 state=started
+#      with_flattened:
+#        - ec2_client.instances
+#        - ec2_servers.instances
+
+- name: Configure roles common to all servers 
+  hosts: client:servers
+  become: True
+  gather_facts: True
+  environment: "{{proxy_env}}"
+  roles:
+    - common
+
+- name: Configure roles specific to jmeter client instance
+  hosts: client
   become: True
-  user: ubuntu
   gather_facts: True
+  environment: "{{proxy_env}}"
   roles:
-    - stress_configure
+    - client 
 
-- name: Launch JMeter slaves
-  hosts: slaves
+- name: Configure roles specific to jmeter server instance(s)
+  hosts: servers
   become: True
-  user: ubuntu
+  gather_facts: True
+  environment: "{{proxy_env}}"
+  roles:
+    - server 
+
+- name: Launch JMeter servers
+  hosts: servers
+  become: True
+  environment: "{{proxy_env}}"
   gather_facts: True
   
   tasks:
-    
     - name: Launch JMeter Server in Background
       # The "JMeter Server" is what the distributed testing nodes run.
-      shell: "( ( nohup /opt/fhir-stress/apache-jmeter-2.13/bin/jmeter-server 1>/dev/null 2>&1 ) & )"
+      shell: "( ( nohup {{ remoteJmeterDir }}/bin/jmeter -s 
+        -Djava.rmi.server.hostname={{ jmeterServerIP }} 
```

Comment:
Minor style issue: Ansible variables should follow Python naming conventions, which I believe use underscores between words instead of camel case, e.g. "`jmeter_server_ip`".

Reply:
Converted all camel case variables to snake case.